src/allotropy/parsers/lines_reader.py

```diff
@@ -36,74 +38,90 @@ def current_line_exists(self) -> bool:
     def get(self) -> Optional[str]:
         return self.lines[self.current_line] if self.current_line_exists() else None
 
-    def match(self, pattern: str) -> bool:
+    def match(self, match_pat: str) -> bool:
         line = self.get()
-        return False if line is None else bool(search(pattern, line))
+        return False if line is None else bool(search(match_pat, line))
 
-    def is_empty(self) -> bool:
-        return self.match("^\\s*$")
+    def is_empty(self, empty_pat: str = EMPTY_STR) -> bool:
```

Comment:
I'm not sold on override the empty string pattern, but if we want to do it, I think  `empty_pattern` should be a field on `LinesReader` and not a parameter on various methods.

Reply:
I hadn't thought about making this a field of the object - I think having it as a parameter results in about the same number of changes to existing code, but am happy to refactor if you prefer the former.